File: Python/Rysowanie GRAFU.py
def convert_to_dot(graph_type, edges):
    if graph_type in {'d', 'dw'}:
        dot_type = 'digraph'
        connector = '->'
    else:
        dot_type = 'graph'
        connector = '--'
    dot_representation = [f"{dot_type} {{"]
    for edge in edges:
        if len(edge) == 3:
            dot_representation.append(f"{edge[0]} {connector} {edge[1]} [label = {edge[2]}];")
        else:
            dot_representation.append(f"{edge[0]} {connector} {edge[1]};")
    dot_representation.append('}')
    return '\n'.join(dot_representation)

graphs = []
t = int(input())
for _ in range(t):
    edges = []
    graph_type = input()
    n = int(input())
    for _ in range(n):
        edge_data = input().split()
        edges.append(edge_data)
    graphs.append((graph_type, edges))
for graph_type, edges in graphs:
    print(convert_to_dot(graph_type, edges), '\n')

File: Java/Rysowanie GRAFU.java
import java.util.List;
import java.util.Scanner;
import java.util.ArrayList;

class Main 
{
    private static String convertToDot(String graphType, List<String[]> edges) 
    {
        String connector = (graphType.equals("d") || graphType.equals("dw")) ? "->" : "--";
        String dotType = (graphType.equals("d") || graphType.equals("dw")) ? "digraph" : "graph";
        StringBuilder result = new StringBuilder(dotType + " {\n");
        for (String[] edge : edges) 
        {
            if (edge.length == 3) 
                result.append("    ").append(edge[0]).append(" ").append(connector).append(" ").append(edge[1]).append(" [label = ").append(edge[2]).append("];\n");
            else 
                result.append("    ").append(edge[0]).append(" ").append(connector).append(" ").append(edge[1]).append(";\n");
        }
        result.append("}");
        return result.toString();
    }

    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int t = input.nextInt();
        input.nextLine();
        List<String> results = new ArrayList<>();
        while (t-- > 0) 
        {
            String graphType = input.nextLine();
            int n = Integer.parseInt(input.nextLine());
            List<String[]> edges = new ArrayList<>();
            for (int i = 0; i < n; ++i) 
            {
                String[] edgeData = input.nextLine().split(" ");
                edges.add(edgeData);
            }
            results.add(convertToDot(graphType, edges));
        }
        for (String res : results)
            System.out.println(res + '\n');
        input.close();
    }
}